src/ngraph/op/topk.hpp

```diff
@@ -96,5 +96,98 @@ namespace ngraph
             virtual void generate_adjoints(autodiff::Adjoints& adjoints,
                                            const NodeVector& deltas) override;
         };
+
+        namespace v1
+        {
+            /// \brief Computes indices and values of the k maximum/minimum values
+            ///        for each slice along specified axis.
+            class TopK : public Op
+            {
+            public:
+                enum class SortType
+                {
+                    NONE,
+                    SORT_INDICES,
+                    SORT_VALUES,
+                };
+
+                enum class Mode
+                {
+                    MAX,
+                    MIN
+                };
+
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"TopK", 1};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                /// \brief Constructs a TopK operation
+                TopK() = default;
+                /// \brief Constructs a TopK operation with two outputs: values and indices.
+                ///        By default the indices output is described by i32 data type.
+                ///
+                /// \param data The input tensor
+                /// \param k Specifies how many maximum/minimum elements should be computed
+                ///          (note: scalar input tensor)
+                /// \param axis The axis along which to compute top k indices
+                /// \param mode Specifies which operation (min or max) is used to select
+                ///             the biggest element of two.
+                /// \param sort Specifies order of output elements and/or indices
+                ///             Accepted values: none, index, value
+                TopK(const Output<Node>& data,
```

Comment:
Sorry, missed this the first time: let's continue to have a constructor that takes the index element type as an attribute.

Reply:
I added `index_element_type` with `i32` as default value - it seems to compromise to provide compatibility with expected interface. @aprocter What do you think about it?